sql/core/src/test/scala/org/apache/spark/sql/internal/SQLConfSuite.scala

```diff
@@ -17,13 +17,36 @@
 
 package org.apache.spark.sql.internal
 
+import org.scalatest.BeforeAndAfterAll
+
 import org.apache.spark.sql.{QueryTest, Row, SparkSession, SQLContext}
+import org.apache.spark.sql.execution.WholeStageCodegenExec
 import org.apache.spark.sql.test.{SharedSQLContext, TestSQLContext}
 
-class SQLConfSuite extends QueryTest with SharedSQLContext {
+class SQLConfSuite extends QueryTest with SharedSQLContext with BeforeAndAfterAll {
```

Comment:
Initially, I tried it. The behavior is controlled by the configuration `SQLConf.MAX_CASES_BRANCHES`. However, I am not sure how to change the default conf value of `SQLConf.MAX_CASES_BRANCHES` in `OptimizeCodegenSuite`. 

At the same time, we do not have a test case to verify the configuration `MAX_CASES_BRANCHES`.

That is why I added the test case here. Let me know if you have any idea. Thanks!